### PR TITLE
Some doc tweaks

### PR DESCRIPTION
**Pull request by fingolfin**
- docs: ensure AbstractAlgebra is available in jldoctests
- docs: upgrade to latest Documenter.jl
- Move a bunch of examples into docstrings
- docs: add a missing space

### CODE REVIEW

**Comment by thofma** [comment]
One namespace error `DocMeta not found` and one error in the doctests.

**Comment by fingolfin** [comment]
Fixed the `DocMeta not found`. But I am really puzzled by the doctest failure.

**Comment by fingolfin** [comment]
Of course I can happily do that, but I still really wonder how that happened? Does anyone have any idea what change I made could have lead to that?

Oh, and really cool: I see that we now get a PR code comment indicating the part in the code with failing test. Nice!

File: docs/src/submodule.md [hunk 1]
```diff
@@ -198,5 +198,5 @@ Domain: Submodule over Integers with 1 generator and no relations
 Codomain: Free module of rank 2 over Integers)
 
 julia> I = intersect(N1, N2)
-Union{AbstractAlgebra.Generic.ModuleHomomorphism{BigInt}, AbstractAlgebra.Generic.Submodule{BigInt}}[]
+Any[]
```
Comment (fingolfin) [inline docs/src/submodule.md:201]: Oh wait, I've changed it here. But why? I think this happened as part of running `doctest(:fix)` or something like that... ?!? Huh. Anyway, if that explains it, then I am happy, phew :-)

Comment (thofma) [reply docs/src/submodule.md]: Maybe you ran `:fix` on a different version?

**Comment by wbhart** [comment]
Yes, that is a nice feature. It saves a lot of work when trying to track down where a failure came from.

File: docs/src/matrix.md [hunk 2]
```diff
@@ -507,31 +507,8 @@ julia> c = M[1, 1]
...
-[  t^2       t             t]
-[   -2   t + 2   t^2 + t + 1]
-
-julia> B = transpose(A)
-[t + 1   t^2            -2]
-[    t     t         t + 2]
-[    1     t   t^2 + t + 1]
-
+```@docs
+transpose(x::Mat{T}) where T <: NCRingElement
```
Comment (fingolfin) [inline docs/src/matrix.md:511]: I think it complains about `Mat` here. Probably need `Generic.Mat` ? But why is it that anyway, and not MatElem / MatrixElem?

Comment (wbhart) [reply docs/src/matrix.md:511]: I don't think it should be `Mat`. I think you are right.

**Review by wbhart — APPROVED** [review]

**Pull request merged**